app/validators/sections/section_validator.py

```diff
@@ -292,3 +296,117 @@ def _get_question_variant_fields_sets(variants):
                 results["number_of_answers"].add(len(results["answer_ids"]))
 
         return results
+
+    def validate_number_of_lists(self):
+        if self.has_list_summary_with_non_item_answers() and self.has_multiple_lists():
+            self.add_error(error_messages.MULTIPLE_LISTS)
+
+    def validate_number_of_list_collectors(self):
+        if (
+            self.has_list_summary_with_non_item_answers()
+            and self.has_multiple_list_collectors()
+        ):
+            self.add_error(error_messages.MULTIPLE_LIST_COLLECTORS)
+
+    def has_list_summary_with_non_item_answers(self):
+        if summary := self.schema_element.get("summary"):
+            show_non_item_answers = summary.get("show_non_item_answers")
+            if summary.get("items") and show_non_item_answers is True:
+                return True
+
+    def has_multiple_list_collectors(self):
+        list_collector_count = 0
+        for group in self.schema_element.get("groups"):
+            for block in group.get("blocks"):
+                if block["type"] in ["ListCollector"]:
+                    list_collector_count += 1
+        if list_collector_count > 1:
+            return True
```

Comment:
Could be simplified with a set comprehension.

Reply:
We need to do this for every group in schema element so I simplified it to list comprehension, same the one above, we only do this for a section not for the whole questionnaire schema.